code/datums/antags/antags.dm

```diff
@@ -0,0 +1,91 @@
+/datum/antag
+	var/name = "Generic Baddie"
+	var/antagtag = "baddie"
+	var/equipped = FALSE
+	var/list/mutually_exclusive = list()
+	var/completely_exclusive = FALSE
+
+	var/ticks = 0
+
+	var/list/datum/objective/objectives = list()
+
+	var/restricted_jobs = list("AI", "Cyborg")
+	var/protected_jobs = list("Security Officer", "Warden", "Detective",
+		"Head of Security", "Captain")
+
+/datum/antag/New()
+	. = ..()
+	if(config.protect_roles_from_antagonist)
+		restricted_jobs += protected_jobs
+
+	if(config.protect_assistant_from_antagonist)
+		restricted_jobs += "Assistant"
+
+/datum/antag/proc/on_gain(datum/mind/user)
+	if(!istype(user))
+		return FALSE
+
+	for(var/i in user.special_roles)
+		var/datum/antag/D = user.special_roles[i]
+		if(completely_exclusive)
+			return FALSE
+		if(D.completely_exclusive)
+			return FALSE
+		if(type in D.mutually_exclusive)
+			return FALSE
+		if(D.type in mutually_exclusive)
+			return FALSE
```

Comment:
For an example of a "completely_exclusive" antag type: blob is completely exclusive with other antag types, as you're literally in an indestructible invisible mobcamera.